doc/sphinx/proof-engine/proof-handling.rst

```diff
@@ -80,7 +80,7 @@ list of assertion commands is given in :ref:`Assertions`. The command
       a while when the proof is large. In some exceptional cases one may
       even incur a memory overflow.
 
-.. cmd:: Defined
+.. cmd:: Defined {? @ident }
```

Comment:
Not sure what you have in mind for rewording.  Seems fine for `Save` and `Defined` to repeat that `@ident` is the theorem name.  Only a few words, and easier to read if it's repeated.

I notice that `Save` overrides the name provided by `Theorem`.  Is that a feature worth documenting or is the lack of a error/warning message an oversight?

Reply:
OK, fine to repeat, but still `Save` should be presented as a variant of `Qed` since it makes the proof opaque like `Qed` and contrary to `Defined`. Yes, when you provide a name to `Save` or `Defined`, you can overwrite the name. This is worth documenting, especially as it has not always been like this (it used to be restricted to `Unamed_thm` names).

Reply:
I notice that `Theorem` takes an `ident_decl`, which allows a `univ_decl` on the theorem name.  That seems incorrect--right?  Perhaps another PR?  And if perchance that is correct, let's hope `Save` keeps the `univ_decl` while only changing the name

Reply:
OK, I put `Save` in the middle.  OTOH, they really should be one command (IMHO `Qed`) with an optional `@ident` and `transparent` flag.  You think that's worth implementing?

Reply:
> I notice that `Theorem` takes an `ident_decl`, which allows a `univ_decl` on the theorem name. That seems incorrect--right?

I don't think this is an error. I believe it has some use cases.

>  And if perchance that is correct, let's hope `Save` keeps the `univ_decl` while only changing the name

Yes, I'm pretty sure it only change the name.

>You think that's worth implementing?

Not worth changing right now because the plan is anyway to change the meaning of `Qed` / `Defined` (separate them from the transparency information). See coq/ceps#42.